File: batch-application/src/main/java/siosio/batchlet/DbAccessBatchlet.java
package siosio.batchlet;

import javax.batch.api.AbstractBatchlet;
import javax.batch.api.BatchProperty;
import javax.enterprise.context.Dependent;
import javax.inject.Inject;
import javax.inject.Named;
import javax.persistence.EntityManager;
import javax.persistence.PersistenceContext;
import javax.transaction.Transactional;
import javax.transaction.Transactional.TxType;

import siosio.entity.UserEntity;

@Dependent
@Named
@Transactional(TxType.REQUIRES_NEW)
public class DbAccessBatchlet extends AbstractBatchlet {

    @PersistenceContext(unitName = "batch")
    EntityManager em;

    @Inject
    @BatchProperty
    String count;

    @Override
    public String process() throws Exception {
        final int insertCount = Integer.parseInt(count);

        for (int i = 0; i < insertCount; i++) {
            em.persist(new UserEntity("name_" + (i + 1)));
        }

        return "SUCCESS";
    }
}

File: dbaccess-batchlet-sample/src/main/java/siosio/DbAccessBatchlet.kt
package siosio

import siosio.sql.*
import javax.batch.api.*
import javax.enterprise.context.*
import javax.inject.*

/**
 * データベース・アクセスを行う[Batchlet]のサンプル実装
 */
@Dependent
@Named
class DbAccessBatchlet : AbstractBatchlet() {

  @Inject
  lateinit var database: Database

  override fun process(): String? {
    database.withTransaction {
      (1..10).forEach {
        database.execute("insert into user (name) values (:name)", User(name = "name_$it"))
      }
    }
    return "success"
  }
}